app/utils/transit_utils.rb

```diff
@@ -1,9 +1,27 @@
 module TransitUtils
+
+  class UUIDWriteHandler
+    def tag(_)
+      "u"
+    end
+
+    def rep(u)
+      Transit::UUID.new(u.to_s)
+    end
+
+    def string_rep(u)
+      u.to_s
+    end
+  end
```

Comment:
To keep the data types symmetrical we must also provide a read handler that returns UUIDTools::UUID objects instead of Transit::UUID. The recursive write handling is an ok simplification.